enclave_manager/avalon_enclave_manager/epid_attestation.py

```diff
@@ -0,0 +1,198 @@
+            'encryption_key_signature':
+                signup_data['encryption_key_signature'],
+            'proof_data': 'Not present',
+            'enclave_persistent_id': 'Not present'
+        }
+
+        # If we are not running in the simulator, we are going to go and get
+        # an attestation verification report for our signup data.
+        if not enclave_info.is_sgx_simulator():
+            logger.info("Running in Intel SGX HW mode")
+            logger.debug("posting verification to IAS")
+            ias_nonce = '{0:032X}'.format(random.getrandbits(128))
+            response = self._ias.post_verify_attestation(
+                quote=signup_data['enclave_quote'], nonce=ias_nonce)
+            logger.debug("posted verification to IAS")
+
+            # check verification report
+            if not self._ias.verify_report_fields(
+                    signup_data['enclave_quote'],
+                    response['verification_report']):
+                logger.debug(
+                    "last error: " + self._ias.last_verification_error())
+                if self._ias.last_verification_error() == "GROUP_OUT_OF_DATE":
+                    logger.warning(
+                        "failure GROUP_OUT_OF_DATE " +
+                        "(update your BIOS/microcode!!!) keep going")
+                else:
+                    logger.error("invalid report fields")
+                    return None
+            # ALL checks have passed
+            logger.info("report fields verified")
+
+            # Now put the proof data into the dictionary
+            signup_info['proof_data'] = json.dumps({
+                'verification_report': response['verification_report'],
+                'ias_report_signature': response['ias_signature'],
+                'ias_report_signing_certificate': response['ias_certificate']
+            })
+            # Grab the EPID pseudonym and put it in the enclave-persistent
+            # ID for the signup info
+            verification_report_dict = \
+                json.loads(response['verification_report'])
+            signup_info['enclave_persistent_id'] = \
+                verification_report_dict.get('epidPseudonym')
+
+        return signup_info
+
+    # -----------------------------------------------------------------
+
+    def init_enclave_info(self, signed_enclave, persisted_sealed_data,
+                          num_of_enclave):
+        """
+        Parameters :
+            @param signed_enclave - The enclave binary read from filesystem
+            @param persisted_sealed_data - persisted sealed data file path
+            @param num_of_enclave - number of enclave to create
+        Returns :
+            @returns True on success False on failure
+        """
+        if not self._tcf_enclave_info:
+            self._tcf_enclave_info = enclave_info.tcf_enclave_info(
+                signed_enclave, self._config['spid'], persisted_sealed_data,
```

Comment:
Is `tcf_enclave_info` specific to EPID based attestation. I can see `self._config['spid']` passed here.

Reply:
Yes it is specific to EPID, in the next PR i will change the name of file as well.

Reply:
Is `tcf_enclave_info` going to be common for EPID and DCAP ? In case,  `tcf_enclave_info`  is specific to EPID let's rename it to something like `tcf_enclave_epid_info`